src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -312,6 +325,58 @@ private void waitForSecurityIndexToBeAtLeastYellow() {
         }
     }
 
+    public void initDefaultConfiguration() {
+        AccessController.doPrivileged((PrivilegedAction<Boolean>) () -> {
+            try {
+                if (!clusterService.state().metadata().hasIndex(securityIndex)) {
+                    initalizeConfigTask.complete(null);
+                    LOGGER.info("Creating index {} and default configs if they are absent", securityIndex);
+                    createSecurityIndexIfAbsent();
+                    uploadConfigurationFiles();
+                    loadConfiguration("Init default configuration on manager node", CType.values());
+                }
+                return true;
+            } catch (Exception e) {
+                LOGGER.error("Cannot apply default config (this is maybe not an error!)", e);
+                return false;
+            }
+        });
+    }
+
+    private void uploadConfigurationFiles() {
+        try {
+            final var configDir = resolveConfigDir();
+            if (Files.exists(CType.CONFIG.configFile(configDir))) {
+                final ThreadContext threadContext = threadPool.getThreadContext();
+                try (StoredContext ctx = threadContext.stashContext()) {
+                    threadContext.putHeader(ConfigConstants.OPENDISTRO_SECURITY_CONF_REQUEST_HEADER, "true");
+                    ConfigHelper.uploadFile(client, CType.CONFIG, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    ConfigHelper.uploadFile(client, CType.ROLES, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    ConfigHelper.uploadFile(client, CType.ROLESMAPPING, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    ConfigHelper.uploadFile(client, CType.INTERNALUSERS, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    ConfigHelper.uploadFile(client, CType.ACTIONGROUPS, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    // FIXME remove it as soon as we will read of version 1 config
```

Comment:
Is there a follow up issue for this comment?

Reply:
AFAIR @cwperks already created an issue for it. I think I will update it.